lib/models/node.js

```diff
@@ -194,6 +204,33 @@ function NodeModelFactory (Model, waterline, _, Promise, Constants, configuratio
                 return next();
             }
             return next();
+        },
+
+        getNodeById: function(nodeId) {
+            var self = this;
+            return self.findOne({id: nodeId}).populate('obms');
+        },
+
+        publishNodeAttrEvent: function (oldNode, newNode, attr) {
```

Comment:
I'm wondering whether it makes more sense to just have this method in the events protocol? There doesn't seem to be a reason to me that it should be a model method, since it doesn't have anything to do with persistence.